numpy/f2py/tests/test_crackfortran.py

```diff
@@ -86,3 +86,26 @@ def test_defaultPublic(self, tmp_path):
         assert 'public' not in mod['vars']['a']['attrspec']
         assert 'private' not in mod['vars']['seta']['attrspec']
         assert 'public' in mod['vars']['seta']['attrspec']
+
+    def test_inlineDefinition(self, tmp_path):
+        f_path = tmp_path / "mod.f90"
+        with f_path.open('w') as ff:
+            ff.write(textwrap.dedent("""\
+            module foo
+              public
+              type, private, bind(c) :: a
```

Comment:
what if this line was 
```
type, bind(c) :: a
``` 
will assert 'public' in tt['a']['attrspec'] still return true, assuming this is a valid statement in fortran ?

Reply:
I've slightly changed the new test to add the `bind(c)` attribute to the `b` type. So it is addressing your concern. If an attribute is given, but without giving the access attribute, the normal rule of applying the default access of the module is applied.